chapter12/sxg/ex_12_2.pl

```diff
@@ -0,0 +1,23 @@
+#!perl
+use warnings;
+use strict;
+
+die "No file names found!\n" unless @ARGV;
+
+my $old_name = shift @ARGV;
+my $old_age = -M $old_name;
+$old_age = 0 if (!defined $old_age) ;

```

Comment:
Could have just initialised `$old_age` to 0 and run through all of ARGV in the foreach I think. The file test and the shifting of ARGV are already part of the loop, so this would cut down the complexity outside of the loop.